src/zenml/cli/artifact.py

```diff
@@ -63,7 +62,13 @@ def list_artifacts(**kwargs: Any) -> None:
 
 
 @artifact.command("delete", help="Delete an artifact.")
-@click.argument("artifact_id")
+@click.argument("artifact_name_or_id")
```

Comment:
When you delete the artifact without specifying a name or id, does it delete all versions by default?

Reply:
Do you think that should be the expected behavior?

Reply:
Right now it only deletes the latest version, but we could easily change it to delete all versions I suppose

Reply:
I think I'd expect it to delete all versions... but not sure. Was more a clarifying question because if it doesn't do that by default, then there should also be a way to make sure it does that I reckon.

Reply:
I would expect full deletion, not only the latest, UX-wise.

Reply:
Okay, changed it to delete all versions if only the name is provided 👍